### DISCUSSION TITLE
crouton troubleshooter: what's wrong and how to fix it

### DISCUSSION THREAD

**Discussion by CroutonIsFun** [dnschneid/crouton#5071] [General]
**What's wrong?**
1. You download crouton, and you try to follow the instructions to make it executable. You pop open a crosh shell and immediately get the warning,

_Sudo commands will not succeed by default. If you want to use sudo commands, use the VT-2 shell_, followed by some other gibberish you don't understand.

If you copy and paste the command,

`sudo install -Dt /usr/local/bin -m 755 ~/Downloads/crouton`,

and press Enter, then you get,

_sudo: The "no new privileges" flag is set, which prevents sudo from running as root.
sudo: If sudo is running in a container, you may need to adjust the container configuration to disable the flag._

Solution: you have to enter the command in VT-2 instead :-(

2. Assuming you manage to get crouton to be executable in VT-2, you then try to install linux, using the command,

`sudo crouton -t xfce`

After watching hundreds of files get installed, and with your heart beating faster because you think, "Finally, I'm getting somewhere", the procedure ends with the following text:

_sed: can't read tests/cras_test_client.c:  No such file or directory
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial..._

**Comment by dnschneid** [comment]
It's super helpful that you're writing up these how-to's, but I think the wiki is a better destination for them rather than being scattered about discussions (since these aren't really discussions).

**Reply by CroutonIsFun** [reply]
@dnschneid 

Thanks! I was wondering about that. Will do!!!
